astroquery/query.py

```diff
@@ -463,3 +496,101 @@ def login(self, *args, **kwargs):
 
     def authenticated(self):
         return self._authenticated
+
+
+class RetrySession(Session):
+    """ Session that automatically does a number of retries for failed
+        transient errors. The time between retries double every time until a
+        maximum of 30sec is reached
+
+        The following network errors are considered transient:
+            requests.codes.unavailable,
+            requests.codes.service_unavailable,
+            requests.codes.gateway_timeout,
+            requests.codes.request_timeout,
+            requests.codes.timeout,
+            requests.codes.precondition_failed,
+            requests.codes.precondition,
+            requests.codes.payment_required,
+            requests.codes.payment
+        """
+
+    retry_errors = [requests.codes.unavailable,
+                    requests.codes.service_unavailable,
+                    requests.codes.gateway_timeout,
+                    requests.codes.request_timeout,
+                    requests.codes.timeout,
+                    requests.codes.precondition_failed,
+                    requests.codes.precondition,
+                    requests.codes.payment_required,
+                    requests.codes.payment]
+
+    def __init__(self, retry=True, start_delay=1, *args, **kwargs):
+        """
+        ::param retry: set to False if retries not required
+        ::param start_delay: start delay interval between retries (default=1s).
+                Note that for HTTP 503, this code follows the retry timeout
+                set by the server in Retry-After
+        """
+        self.retry = retry
+        self.start_delay = start_delay
+        super(RetrySession, self).__init__(*args, **kwargs)
```

Comment:
this can be simplified now that the old python versions are dropped.
```suggestion
        super().__init__(*args, **kwargs)
```